v2/src/features/UploadList/types.ts

```diff
@@ -0,0 +1,39 @@
+export type ErrorState = null | 'retriable' | 'non-retriable';
+
+export type Action =
+  | { type: 'TRY_UPLOAD'; payload: { file: File; uuid: string } }
+  | { type: 'CANCEL_UPLOAD'; payload: { uuid: string } }
+  | {
+      type: 'SET_SHAREABLE_LINK';
+      payload: { uuid: string; shareableLink: string };
+    }
+  | {
+      type: 'SET_ERROR';
+      payload: {
+        uuid: string;
+        error: {
+          state: ErrorState;
+          text: string;
+          hint?: string;
+        };
+      };
+    }
+  | {
+      type: 'UPDATE_PROGRESS';
+      payload: { uuid: string; progress: number };
+    };
+
+export type SugarFileState = {
+  file: File;
+  shareableLink: string;
+  uuid: string;
+  upload: {
+    state: boolean;
```

Comment:
Do we know what this could be? Maybe better to say explicitly with string values like `ongoing`, `aborted`, ...

Reply:
Issue created to address this separatedly